clients/apps/web/src/components/Subscriptions/UpdateSubscriptionModal.tsx

```diff
@@ -576,6 +577,113 @@ const UpdateTrial = ({
   )
 }
 
+const UpdateBillingPeriod = ({
+  subscription,
+  onUpdate,
+}: {
+  subscription: schemas['Subscription']
+  onUpdate?: () => void
+}) => {
+  const updateSubscription = useUpdateSubscription(subscription.id)
+
+  const minDate = useMemo<Date | undefined>(() => {
+    if (subscription.current_period_end) {
+      return new Date(subscription.current_period_end)
+    }
+    return new Date()
+  }, [subscription])
+
+  const form = useForm<schemas['SubscriptionUpdateBillingPeriod']>({
+    defaultValues: {
+      current_billing_period_end: subscription.current_period_end || undefined,
+    },
+  })
+  const { control, handleSubmit, setError } = form
+
+  const onSubmit = useCallback(
+    async (body: schemas['SubscriptionUpdateBillingPeriod']) => {
+      await updateSubscription.mutateAsync(body).then(({ error }) => {
+        if (error) {
+          if (error.detail)
+            if (isValidationError(error.detail)) {
+              setValidationErrors(
+                error.detail,
+                setError,
+                undefined,
+                validationDiscriminators,
+              )
+            } else {
+              toast({
+                title: 'Billing period update failed',
+                description: `Error while updating billing period for ${subscription.product.name}: ${error.detail}`,
+              })
+            }
+          return
+        }
+
+        toast({
+          title: 'Billing period updated',
+          description: `Billing period for ${subscription.product.name} has been successfully updated`,
+        })
+        onUpdate?.()
+      })
+    },
+    [updateSubscription, subscription, setError, onUpdate],
+  )
+
+  return (
+    <div className="dark:bg-polar-800 flex flex-col gap-y-4 rounded-2xl bg-gray-50 p-6">
+      <div className="flex flex-col gap-y-2">
+        <h3 className="text-lg font-medium">Update Billing Period</h3>
+        <p className="dark:text-polar-500 mt-1 text-sm text-gray-500">
+          Extend the current billing period by setting a new end date in the
+          future. This is useful for providing additional subscription time to a
```

Comment:
Minor copy suggestion (feel free to discard): "for providing additional _free_ subscription time"

Reply:
Thanks!